protocol-specs/generators/dpos_random_seed_generation/index.js

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { hash } = require('@liskhq/lisk-cryptography');
+
+const BaseGenerator = require('../base_generator');
+const { list: sampleDelegateList } = require('./forger_list');
+
+/* eslint-disable new-cap, no-continue */
+
+const activeDelegates = 101;
+const standByDelegates = 2;
+
+const strippedHash = data => {
+	let input;
+
+	if (typeof data === 'number') {
+		input = Buffer.from(data.toString(16), 'hex');
+	} else if (typeof data === 'string') {
+		input = Buffer.from(data, 'hex');
+	} else if (data instanceof Buffer) {
+		input = data;
+	} else {
+		throw new Error('Hash input is not a valid type');
+	}
+
+	return hash(input)
+		.slice(0, 16)
+		.toString('hex');
```

Comment:
can we handle this as Buffer instead of converting to hex string?

Reply:
We always have to use it as a string, in seed reveal or random seed. So it's convenient to use hex string output.

Reply:
I think this protocol spec especially, it's doing the bitwise operation. It make sense to do it consistently with buffer. `hex` is just a representation. Also, in this file, it's used as buffer mainly anyway